src/gui/qgslockbutton.h

```diff
@@ -0,0 +1,78 @@
+/***************************************************************************
+    qgslockbutton.cpp - Lock button
+     --------------------------------------
+    Date                 : July, 2017
+    Copyright            : (C) 2017 by Mathieu Pellerin
+    Email                : nirvn dot asia at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSLOCKBUTTON_H
+#define QGSLOCKBUTTON_H
+
+#include "qgslockbutton.h"
+
+#include <QToolButton>
+#include "qgis_gui.h"
+#include "qgis.h"
+
+/** \ingroup gui
+ * \class QgsLockButton
+ * A cross platform button subclass used to represent a locked / unlocked state.
+ * \since QGIS 3.0
+ */
+class GUI_EXPORT QgsLockButton : public QToolButton
```

Comment:
I think QgsRatioLockButton might be more suitable. This button is really designed for that purpose alone and includes the custom drawing to handle that. I'd rather keep this class specific so that it's not used for other purposes in future and prevents it focusing on that single purpose.